### PR TITLE
Gha pipeline migration

### PR DESCRIPTION
**Pull request by ChrisScottThomas**
Migration from Concourse to GitHub Actions.

- removes Concourse pipelines
- removes scripts associated with Concourse pipelines

- Adds GHA workflows
  -  when creating a pull-request with Terraform changes
  -  when creating a pull-request with NON-Terraform changes
  -  when pushing Terraform changes, to any branch but Master
  -  when pushing (merge) Terraform changes to Master

Note:
> Applying to Management is currently hashed out due to work yet to be completed by third-party, which if applied, will break Management.

### CODE REVIEW

File: .github/workflows/tf-merge.yml [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        run: terraform init
+        env:
+          TF_VAR_assume_role: gha_aws_concourse
+          AWS_ACCESS_KEY_ID: ${{ secrets.ACTIONS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.ACTIONS_SECRET_ACCESS_KEY }}
+          AWS_DEFAULT_REGION: eu-west-2
+
+      - name: "Terraform Apply"
+        id: plan
+        run: terraform apply -auto-approve
```
Comment (dan-hill2802) [inline .github/workflows/tf-merge.yml:57]: can you make this follow the other pattern, output a plan to file then apply that plan?

File: .github/workflows/tf-merge.yml [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        env:
+          TF_WORKSPACE: default
+          TF_VAR_assume_role: gha_aws_concourse
+          AWS_ACCESS_KEY_ID: ${{ secrets.ACTIONS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.ACTIONS_SECRET_ACCESS_KEY }}
+          AWS_DEFAULT_REGION: eu-west-2
+
+      - name: "Terraform Plan"
+        id: plan
+        run: terraform plan
```
Comment (dan-hill2802) [inline .github/workflows/tf-merge.yml:68]: needs a detailed exit code

File: .github/workflows/tf-merge.yml [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+      #   env:
+      #     TF_WORKSPACE: management
+      #     TF_VAR_assume_role: gha_aws_concourse
+      #     AWS_ACCESS_KEY_ID: ${{ secrets.ACTIONS_ACCESS_KEY_ID }}
+      #     AWS_SECRET_ACCESS_KEY: ${{ secrets.ACTIONS_SECRET_ACCESS_KEY }}
+      #     AWS_DEFAULT_REGION: eu-west-2
+
+      - name: "Terraform Plan"
+        id: plan
+        run: terraform plan
```
Comment (dan-hill2802) [inline .github/workflows/tf-merge.yml:133]: needs detailed exit code

File: .github/workflows/tf-merge.yml [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        run: terraform init
+        env:
+          TF_VAR_assume_role: gha_aws_concourse
+          AWS_ACCESS_KEY_ID: ${{ secrets.ACTIONS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.ACTIONS_SECRET_ACCESS_KEY }}
+          AWS_DEFAULT_REGION: eu-west-2
+
+      # - name: "Terraform Apply"
+      #   id: plan
+      #   run: terraform apply -auto-approve
```
Comment (dan-hill2802) [inline .github/workflows/tf-merge.yml:122]: this should plan to output file then use it in the apply

File: .github/workflows/tf-merge.yml [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+        working-directory: .
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.ACTIONS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.ACTIONS_SECRET_ACCESS_KEY }}
+          AWS_SECRETS_ROLE: ${{ secrets.AWS_GHA_ROLE_MGMT_DEV }}
+          AWS_DEFAULT_REGION: eu-west-2
+
+      - name: "Terraform Format"
+        id: fmt
+        run: terraform fmt
```
Comment (dan-hill2802) [inline .github/workflows/tf-merge.yml:43]: I'm not entirely sure why fmt is being run, but if needed then it should have --recursive

**Review by mattburgess — APPROVED** [review]

**Pull request merged**